src/resources/dart-3-migration.md

```diff
@@ -0,0 +1,545 @@
+../pkg1/lib/pkg1.dart:1:1: Error: Library doesn't support null safety.
+// @dart=2.9
+^^^^^^^^^^^^
+```
+
+#### Migration
+
+Before beginning any migration to Dart 3, ensure your app or package has been 100% migrated to enable null safety. 
+This requires a Dart `2.19` SDK, not a Dart 3 SDK. For details, see the [null safety migration guide](/null-safety/migration-guide).
+
+### Colon-syntax for default values
+
+For historical reasons, named optional parameters could specify their default
+value using either `:` or `=`. In Dart 3, only the `=` syntax is allowed.
+
+#### Scope
+
+This is an [*unversioned* change](#unversioned-vs-versioned-changes), that applies to all Dart 3 code.
+
+#### Symptom
+
+Dart analysis produces errors like:
+
+```
+line 2 • Using a colon as a separator before a default value is no longer supported.
+```
+
+#### Migration
+
+Change from using colons:
+
+```dart
+int someInt({int x: 0}) => x;
+```
+
+To using equals:
+
+```dart
+int someInt({int x = 0}) => x;
+```
+
+This migration can be made manually, or automated with `dart fix`:
+
+```
+$ dart fix --apply --code=obsolete_colon_for_default_value
+```
+
+### `mixin`
+
+Pre-Dart 3, any `class` could be used as a `mixin`,
+as long as it had no declared constructors and no superclass other than `Object`.
+
+In Dart 3, classes declared in libraries at language version 3.0 or later
+can't be used as mixins unless marked `mixin`.
+This restriction applies to code in any library attempting to use the class as a mixin, 
+regardless of the latter library's language version.
+
+#### Scope
+
+This is a [*versioned* change](#unversioned-vs-versioned-changes), that only applies to language version 3.0 or later.
+
+#### Symptom
+
+An analysis error like:
+
+```
+Mixin can only be applied to class.
+```
+
+The analyzer produces this diagnostic when a class that is neither a
+`mixin class` nor a `mixin` is used in a `with` clause.
+
+#### Migration
+
+Determine if the class is intended to be used as a mixin.
+
+If the class defines an interface, consider using `implements`.
+
+### `switch`
+
+Dart 3.0 interprets [switch cases](/language/control-flow#switch-and-case)
+as patterns [*TODO* LINK] instead of constant expressions. 
+
+#### Scope
+
+This is a [*versioned* change](#unversioned-vs-versioned-changes), that only applies to language version 3.0 or later.
+
+#### Symptom
+
+Most constant expressions found in switch cases are valid patterns
+with the same meaning (named constants, literals, etc.).
+These will behave the same and no symptoms will arise.
+
+The few constant expressions that aren't valid patterns will trigger the [`invalid_case_patterns` lint](/tools/linter-rules#invalid_case_patterns).
+
+#### Migration
+
+You can revert back to the original behavior by prefixing the case pattern with `const`, so it's no longer interpreted as a pattern:
+
+{enclose in backticks:}
+case const [1, 2]:
+case const {'k': 'v'}:
+case const {1, 2}:
+case const Point(1, 2):
```

Comment:
Oh sorry, that wasn't supposed to go in the text, I wrote that note in for you to add the backticks because if I added ``` in the suggestion box, it doesn't render and ends the box

```suggestion
case const [1, 2]:
case const {'k': 'v'}:
case const {1, 2}:
case const Point(1, 2):
```